### PR TITLE
Fix build of library on recent Cygwin editions

### PR DESCRIPTION
**Pull request by jeking3**
Without this fix you get the behavior in #144.

### CODE REVIEW

**Comment by raffienficiaud** [comment]
On the branch ``topic/PR-145-cygwin-fixes``, tests cycling.

File: build/Jamfile.v2 [hunk 1]
```diff
@@ -12,6 +14,7 @@ project boost/test
     : source-location ../src
     : requirements <link>shared:<define>BOOST_TEST_DYN_LINK=1
                    <toolset>borland:<cxxflags>-w-8080
+                   <target-os>cygwin:<define>_POSIX_C_SOURCE=201112L
```
Comment (jeking3) [inline build/Jamfile.v2]: This is the same fix I put into the appveyor.yml of all affected repositories that use the boost-ci framework.

https://github.com/jeking3/boost-ci/blob/master/templates/appveyor.yml

The only problem here is that the value should be 200112L and not 201112L

**Comment by jeking3** [comment]
Also do you want a full CI environment (same that now exists on pool, ptr_container, format, uuid, ...?)

**Comment by raffienficiaud** [comment]
Why not, I have my internal CI on OSX/Linux/Win, and on Win only different versions of Visual. Cygwin might be good to have as well.

File: build/Jamfile.v2 (hunk 1 above)
Comment (raffienficiaud) [reply build/Jamfile.v2]: So what should I do for the bjam? `200112L` or `20112L` ?

**Comment by jeking3** [comment]
The point of putting CI here is to allow people to self-serve their own quality of pull requests and to reduce the amount of work it takes for your to qualify a change.

I pulled this PR and a CI PR into my fork:

https://github.com/jeking3/test/pull/1

Appveyor: https://ci.appveyor.com/project/jeking3/test/build/1.0.6-develop
Travis CI: https://travis-ci.org/jeking3/test/builds/394392360

I haven't looked into the failures yet.  Regardless, this particular fix (in this PR) is what I have used to resolve cygwin build issues in the repositories I maintain and in the CMT repositories.  In those, the definition is performed in the appveyor script so that Boost.Test builds.

**Comment by raffienficiaud** [comment]
You are referring to a CI that is well integrated to GitHub. In all cases, I am running my own tests on the feature branches before merging, the configuration is stable for some years. Said that, you are right that it is always extra work to add a new platform, and I thank you for the ones that you are currently deploying. Let me know once it is stable on your end.

**Comment by raffienficiaud** [comment]
Going back to the purpose of the PR: the diff of this is `<target-os>cygwin:<define>_POSIX_C_SOURCE=201112L`, while the configuration you have on your Appveyor is `<target-os>cygwin:<define>_POSIX_C_SOURCE=2001112L`. At the end, I am confused on which one to take, my CI is blind to this change, yours is not green

**Comment by raffienficiaud** [comment]
Hi again,

I would like to merge this branch asap. Would you please confirm which of the options is correct? `201112L` or `2001112L` ?

**Comment by jeking3** [comment]
2001 is correct

**Comment by raffienficiaud** [comment]
I just tried on a fresh install of cygwin, and with or without the patch, the code compiles without problem. Just wondering how to reproduce. Any hint?

**Comment by raffienficiaud** [comment]
Closing, merged to master/1.68.

**Pull request closed without merging**